Mathlib/Topology/ExtremallyDisconnected.lean

```diff
@@ -139,8 +134,163 @@ protected theorem CompactT2.Projective.extremallyDisconnected [CompactSpace X] [
+noncomputable def ExtremallyDisconnected.homeoCompactToT2 [ExtremallyDisconnected A] [T2Space A]
+    [T2Space E] [CompactSpace E] {ρ : E → A} (ρ_cont : Continuous ρ) (ρ_surj : ρ.Surjective)
+    (zorn_subset : ∀ E₀ : Set E, E₀ ≠ univ → IsClosed E₀ → ρ '' E₀ ≠ univ) : E ≃ₜ A :=
+  ρ_cont.homeoOfEquivCompactToT2
+    (f := Equiv.ofBijective ρ ⟨homeoCompactToT2_injective ρ_cont ρ_surj zorn_subset, ρ_surj⟩)
+
+/-- Theorem 2.5 in [Gleason, *Projective topological spaces*][gleason1958]:
+in the category of compact spaces and continuous maps,
+the projective spaces are precisely the extremally disconnected spaces.
+
+Proof. Let $A$ be an extremally disconnected compact space, let $B$ and $C$ be compact spaces, and
+let $f : B \twoheadrightarrow C$ and $\phi : A \to C$ be continuous maps. Consider the compact space
+$D := \{(a, b) : \phi(a) = f(b)\}$ with projections $\pi_1 : D \to A$ and $\pi_2 : D \to B$, which
+has a closed subset $E$ satisfying the "Zorn subset condition" by Lemma 2.4, such that $\pi_1|_E$
+is a homeomorphism by Lemma 2.3. Then $\phi = f \circ \pi_2|_E \circ \pi_1|_E^{-1}$. -/
+protected theorem CompactT2.ExtremallyDisconnected.projective [ExtremallyDisconnected A]
+    [CompactSpace A] [T2Space A] : CompactT2.Projective A := by
+  intro B C _ _ _ _ _ _ φ f φ_cont f_cont f_surj
+  let D : Set <| A × B := {x | φ x.fst = f x.snd}
+  have D_comp : CompactSpace D := isCompact_iff_compactSpace.mp
+    (isClosed_eq (φ_cont.comp continuous_fst) (f_cont.comp continuous_snd)).isCompact
+  let π₁ : D → A := Prod.fst ∘ Subtype.val
+  have π₁_cont : Continuous π₁ := continuous_fst.comp continuous_subtype_val
+  have π₁_surj : π₁.Surjective := fun a => ⟨⟨⟨a, _⟩, (f_surj <| φ a).choose_spec.symm⟩, rfl⟩
+  rcases exists_compact_surjective_zorn_subset π₁_cont π₁_surj with ⟨E, _, E_onto, E_min⟩
+  let ρ : E → A := E.restrict π₁
+  have ρ_cont : Continuous ρ := π₁_cont.continuousOn.restrict
+  have ρ_surj : ρ.Surjective := fun a => by
+    rcases (E_onto ▸ mem_univ a : a ∈ π₁ '' E) with ⟨d, ⟨hd, rfl⟩⟩; exact ⟨⟨d, hd⟩, rfl⟩
+  let ρ' := ExtremallyDisconnected.homeoCompactToT2 ρ_cont ρ_surj E_min
+  let π₂ : D → B := Prod.snd ∘ Subtype.val
+  have π₂_cont : Continuous π₂ := continuous_snd.comp continuous_subtype_val
+  refine ⟨E.restrict π₂ ∘ ρ'.symm, ⟨π₂_cont.continuousOn.restrict.comp ρ'.symm.continuous, ?_⟩⟩
+  suffices f ∘ E.restrict π₂ = φ ∘ ρ' by
+    rw [← comp.assoc, this, comp.assoc, Homeomorph.self_comp_symm, comp.right_id]
+  ext x
+  exact x.val.mem.symm
+
+protected theorem CompactT2.projective_iff_extremallyDisconnnected [CompactSpace A] [T2Space A] :
+    Projective A ↔ ExtremallyDisconnected A :=
+  ⟨Projective.extremallyDisconnected, fun _ => ExtremallyDisconnected.projective⟩
+
+end
+
 -- Note: It might be possible to use Gleason for this instead
-/-- The sigma-type of extremally disconneted spaces is extremally disconnected -/
+/-- The sigma-type of extremally disconnected spaces is extremally disconnected. -/
```

Comment:
Have you tried to implement the suggestion in this note?

Reply:
No, I have not. Presumably Dagur wanted to say we can first prove the sigma type of projective spaces is projective, then reduce the same property for extremally disconnected spaces to this. I can add this,

Reply:
... which is not true unless we add the assumption of compact Hausdorff - is this desirable?